hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -228,9 +251,16 @@ public void promoteSuspended() {
         for (ScheduledTaskDescriptor descriptor : tasks.values()) {
             try {
                 log(FINEST, descriptor.getDefinition().getName(), "Attempting promotion");
-                if (descriptor.shouldSchedule()) {
+                boolean wasActive = descriptor.isActive();
+                if (descriptor.canBeScheduled()) {
                     doSchedule(descriptor);
                 }
+
+                if (!wasActive) {
+                    acquirePermit(true);
+                }
+
+                descriptor.setActive();
```

Comment:
Doesn't `doSchedule(descriptor);` do the same thing in `descriptor.setScheduledFuture(future);`?